### PR TITLE
Homework 7

### PR DESCRIPTION
**Pull request by beaufsichtigen**

### CODE REVIEW

File: src/main/java/com/epam/tc/hw7/entities/User.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.epam.tc.hw7.entities;
+
+public class User {
+    public static final User ROMAN = new User("Roman", "Jdi1234", "ROMAN IOVLEV");
+
+    private String name;
+    private String password;
+    private String fullName;
```
Comment (shoggy) [inline src/main/java/com/epam/tc/hw7/entities/User.java:8]: As improvement, please make these fields `final`

File: src/main/java/com/epam/tc/hw7/entities/User.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.epam.tc.hw7.entities;
+
+public class User {
+    public static final User ROMAN = new User("Roman", "Jdi1234", "ROMAN IOVLEV");
```
Comment (shoggy) [inline src/main/java/com/epam/tc/hw7/entities/User.java:4]: It is not good practice for DTO to store self constants
Please move it in other place

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/data/DataProviderJson.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                            new TypeReference<HashMap<String, MetalsInfo>>() {
+                            });
+            int i = 0;
+            object = new Object[map1.size()][1];
+            for (Map.Entry<String, MetalsInfo> entry : map1.entrySet()) {
+                object[i][0] = entry.getValue();
+                i++;
+            }
+        } catch (IOException e) {
+            System.out.println("Error while reading .json data file: " + e);
```
[the remark this answers is not in the document]
Comment (beaufsichtigen) [reply src/test/java/com/epam/tc/hw7/data/DataProviderJson.java:30]: Fixed

File: src/test/java/com/epam/tc/hw7/ExerciseTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public void setUp() {
+        PageFactory.initSite(JdiSite.class);
+    }
+
+    @AfterSuite(alwaysRun = true)
+    public void teardown() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test(priority = 0)
```
[the remark this answers is not in the document]
Comment (beaufsichtigen) [reply src/test/java/com/epam/tc/hw7/ExerciseTest.java:27]: When debugging, the test with parameters was launched before login and then crashed. Now it is no longer reproduced. Priorities deleted.

File: src/main/java/com/epam/tc/hw7/pages/JdiMetalsPage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    public MetalColorsForm metalColorsForm;
+
+    @Css(".results li")
+    private WebList log;
+
+    public void fillMetalColorForm(MetalsInfo metalInfo) {
+        metalColorsForm.fill(metalInfo);
+    }
+
+    public List<String> getExpectedLog() {
```
[the remark this answers is not in the document]
Comment (beaufsichtigen) [reply src/main/java/com/epam/tc/hw7/pages/JdiMetalsPage.java:29]: Rename methods here to getActualLogFromPage() 
and in MatalsInfo to getExpectedLogFromJson()  
Assert corrected as well.

File: src/main/java/com/epam/tc/hw7/entities/User.java (hunk 1 above)
Comment (beaufsichtigen) [reply src/main/java/com/epam/tc/hw7/entities/User.java:8]: Done

File: src/main/java/com/epam/tc/hw7/entities/User.java (hunk 2 above)
Comment (beaufsichtigen) [reply src/main/java/com/epam/tc/hw7/entities/User.java:4]: Fixed with GetProperties Utility

**Review by shoggy — APPROVED** [review]

**Pull request merged**